libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -400,8 +428,15 @@ public static RestRequest getRequestForUpdate(String apiVersion, String objectTy
 	 * @throws IOException 
 	 */
 	public static RestRequest getRequestForUpsert(String apiVersion, String objectType, String externalIdField, String externalId, Map<String, Object> fields) throws IOException  {
-    	return new RestRequest(RestMethod.PATCH, RestAction.UPSERT.getPath(apiVersion, objectType, externalIdField, externalId), fields == null ? null : new JSONObject(fields));
-	}
+        return new RestRequest(
```

Comment:
In v37.0, they enhanced upsert to allow null id (as a way to do a create).

Added support in RestRequest for specifying if-unmodified-since header during update

Initially I wanted to use it during overwrite type sync up when a locally update record was remotely deleted and therefore needs to be recreated - but it was not a good fit (if you pass and id it must exists, it only does a create when passed null, I wanted to pass an id and have a create if it doesn't exist).